### PR TITLE
v0.9.1

### PR DESCRIPTION
**Pull request by johnBartos**
https://github.com/video-dev/hls.js/releases/tag/v0.9.0

This brings us pretty close to upstream master:
https://github.com/video-dev/hls.js/compare/master...jwplayer:feature/v0.9.0

### CODE REVIEW

**Comment by johnBartos** [comment]
Broken: http://b028.wpc.azureedge.net/80B028/Samples/a38e6323-95e9-4f1f-9b38-75eba91704e4/5f2ce531-d508-49fb-8152-647eba422aec.ism/Manifest(format=m3u8-aapl)

**Comment by johnBartos** [comment]
test this please

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+module.exports = {
```
Comment (robwalch) [inline .eslintrc.js]: Should we remove the double quotes (`eslint .eslintrc.js --fix`)?

**Comment by robwalch** [comment]
[:building_construction: jwplayer-commercial](https://jenkins.jwplayer.com/view/JW7/job/jw7-commercial/) will now always rebuild dist files for PRs made in this project. Added this to that job's build script before our player build so dist files are updated:

```
if [ ! -z $hlsjsGitBranch ]; then
	echo "\n==== BUILD HLS.JS @ $hlsjsGitBranch ===="
	grunt build-hls.js
fi
```

We still want updated dist files for development and release builds. That task has install dependencies and build hls.js so it's slow.

**Comment by robwalch** [comment]
Are these changes closer to v0.9.1?
```
Changing hls.js : git+ssh://git@github.com/jwplayer/hls.js.git#refs/heads/master
      to hls.js : git+ssh://git@github.com/jwplayer/hls.js.git#102590194bfd398638acbf1e3ea746039a4a874a
warning " > eslint-loader@1.9.0" has unmet peer dependency "eslint@>=1.6.0 <5.0.0".
warning " > eslint-plugin-no-for-of-loops@1.0.0" has unmet peer dependency "eslint@>=2.0.0".
warning " > grunt-contrib-connect@1.0.2" has unmet peer dependency "grunt@>=0.4.0".
warning " > grunt-karma@2.0.0" has unmet peer dependency "grunt@>=0.4.x".
warning " > grunt-postcss@0.9.0" has unmet peer dependency "grunt@>=0.4.5".
├─ hls.js@0.9.1
└─ url-toolkit@2.1.4
```

File: src/loader/playlist-loader.js [hunk 2]
```diff
@@ -1,632 +1,502 @@
...
+      // details = ...?
+      fatal = false;
     }
+
     if (loader) {
       loader.abort();
-      this.loaders[context.type] = undefined;
+      this.resetInternalLoader(context.type);
     }
-    this.hls.trigger(Event.ERROR, {type: ErrorTypes.NETWORK_ERROR, details: details, fatal: fatal, url: context.url, loader: loader, response: response, context : context, networkDetails: networkDetails});
```
Comment (robwalch) [inline src/loader/playlist-loader.js]: `{ code, text } =  response` is expected as a property of `NETWORK_ERROR` type `ERROR` events, but it's been removed here.

File: src/loader/playlist-loader.js [hunk 3]
```diff
@@ -1,632 +1,503 @@
...
-    if (avcdata.length > 2) {
-      result = avcdata.shift() + '.';
-      result += parseInt(avcdata.shift()).toString(16);
-      result += ('000' + parseInt(avcdata.shift()).toString(16)).substr(-4);
-    } else {
-      result = codec;
-    }
-    return result;
+  loadtimeout (stats, context, networkDetails = null) {
+    this._handleNetworkError(context, networkDetails, true);
```
Comment (robwalch) [inline src/loader/playlist-loader.js]: @johnBartos  `_handleNetworkError` is also called here.

**Pull request merged**